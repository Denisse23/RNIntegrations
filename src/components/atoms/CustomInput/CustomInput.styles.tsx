import styled from '@emotion/native';
import { ViewProps } from 'react-native';
import { CustomText } from '@components/atoms';
import { ThemeType } from '@theme/Themes';
import { scaleBySize } from '@utils/Scale';
import { applyOpacityToHex } from '@utils/Opacity';
import { CustomInputProps } from './CustomInput';

export const StyledContainer = styled.View(
  ({
    theme,
    error,
  }: ViewProps & {
    theme: ThemeType;
    error: CustomInputProps['error'];
  }) => ({
    height: scaleBySize(50),
    paddingHorizontal: theme.spacings.lg,
    borderRadius: theme.borderRadius.lg,
    borderTopWidth: scaleBySize(3),
    borderLeftWidth: scaleBySize(3),
    borderBottomWidth: scaleBySize(1),
    borderRightWidth: scaleBySize(1),
    borderColor: error ? theme.colors.danger400 : theme.colors.background200,
    backgroundColor: applyOpacityToHex(
      theme.colors.background200,
      theme.opacities[100],
    ),
  }),
);

export const StyledInput = styled.TextInput(({ theme, value }) => ({
  position: 'absolute',
  height: '100%',
  width: '100%',
  paddingTop: value ? scaleBySize(theme.spacings.lg) : 0,
  paddingHorizontal: scaleBySize(theme.spacings.lg),
  fontSize: theme.fontSizes.md,
  color: theme.colors.text300,
}));

export const StyledCustomText = styled(CustomText)(({ theme }) => ({
  marginTop: scaleBySize(theme.spacings.md),
  marginLeft: scaleBySize(theme.spacings.sm),
  color: theme.colors.text200,
}));

export const StyledErrorCustomText = styled(CustomText)(({ theme }) => ({
  marginTop: scaleBySize(theme.spacings.md),
  marginLeft: scaleBySize(theme.spacings.lg),
  color: theme.colors.danger400,
}));
