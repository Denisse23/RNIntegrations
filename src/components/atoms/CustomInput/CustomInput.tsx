import React from 'react';
import { TextInputProps } from 'react-native';
import { useTheme } from '@emotion/react';
import {
  StyledInput,
  StyledContainer,
  StyledCustomText,
} from './CustomInput.styles';

interface CustomInputProps extends TextInputProps {
  label?: string;
}

const CustomInput = ({
  label,
  placeholder,
  value,
  onChangeText,
  ...rest
}: CustomInputProps) => {
  const theme = useTheme();

  return (
    <StyledContainer>
      {value && <StyledCustomText>{label}</StyledCustomText>}
      <StyledInput
        placeholder={label ? label : placeholder}
        placeholderTextColor={theme.colors.text200}
        value={value}
        onChangeText={onChangeText}
        {...rest}
      />
    </StyledContainer>
  );
};

export default CustomInput;
