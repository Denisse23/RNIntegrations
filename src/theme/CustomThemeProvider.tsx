import React from 'react';
import { useColorScheme } from 'react-native';
import { ThemeProvider } from '@emotion/react';
import { ColorScheme, DarkTheme, LightTheme } from './Themes';

const CustomThemeProvider = ({ children }: { children: React.ReactNode }) => {
  const scheme = useColorScheme();

  return (
    <ThemeProvider theme={scheme === ColorScheme.DARK ? DarkTheme : LightTheme}>
      {children}
    </ThemeProvider>
  );
};

export default CustomThemeProvider;
