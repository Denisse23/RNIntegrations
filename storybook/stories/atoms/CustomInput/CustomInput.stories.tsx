import React from 'react';
import { CustomInput } from '@components/atoms';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import CustomThemeProvider from '@theme/CustomThemeProvider';
import CenterView from '../../CenterView';

storiesOf('CustomInput', module)
  .addDecorator(story => <CenterView>{story()}</CenterView>)
  .add('normal', () => (
    <CustomThemeProvider>
      <CustomInput label={text('Text', 'Normal')} />
    </CustomThemeProvider>
  ))
  .add('with error', () => (
    <CustomThemeProvider>
      <CustomInput
        label={text('Text', 'With error')}
        error={text('Text', 'Error label.')}
      />
    </CustomThemeProvider>
  ));
