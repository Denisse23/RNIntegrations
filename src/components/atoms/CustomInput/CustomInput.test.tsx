import React from 'react';
import renderer from 'react-test-renderer';
import CustomInput, { CustomInputProps } from './CustomInput';
import { INPUT_COMMON_PROPS } from '@constants';
import CustomThemeProvider from '@theme/CustomThemeProvider';

interface CustomInputTestObj extends CustomInputProps {
  incorrectText: string;
  correctText: string;
}

const customInputTests = [
  {
    testID: 'input-email',
    label: 'Email',
    incorrectText: 'email@domain@domain.com',
    correctText: 'email@domain.com',
    ...INPUT_COMMON_PROPS.email,
  },
  {
    label: 'Name',
    incorrectText: 'Test1234',
    correctText: 'Test',
    ...INPUT_COMMON_PROPS.name,
  },
  {
    label: 'New password',
    incorrectText: 'Test234',
    correctText: 'Test12345!',
    ...INPUT_COMMON_PROPS.newPassword,
  },
] as CustomInputTestObj[];

describe('CustomInput', () => {
  let customInputTest = (props: CustomInputTestObj) => {
    it(`${props.label} incorrect should match the snapshot`, () => {
      const tree = renderer.create(
        <CustomThemeProvider>
          <CustomInput {...props} value={props.incorrectText} />
        </CustomThemeProvider>,
      );
      expect(tree).toMatchSnapshot();
    });

    it(`${props.label} correct should match the snapshot`, () => {
      const tree = renderer.create(
        <CustomThemeProvider>
          <CustomInput {...props} value={props.correctText} />
        </CustomThemeProvider>,
      );
      expect(tree).toMatchSnapshot();
    });
  };

  customInputTests.forEach(input => {
    customInputTest(input);
  });
});
