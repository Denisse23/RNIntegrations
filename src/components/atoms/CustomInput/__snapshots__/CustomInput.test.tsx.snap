// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput Email correct should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#D5D5D533",
        "borderBottomWidth": 2.34375,
        "borderColor": "#D5D5D5",
        "borderLeftWidth": 7.03125,
        "borderRadius": 75,
        "borderRightWidth": 2.34375,
        "borderTopWidth": 7.03125,
        "height": 117.1875,
        "paddingHorizontal": 18.75,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 28.125,
          "fontWeight": "400",
        },
        Array [
          Object {
            "color": "#A9A8A3",
            "marginLeft": 10.986328125,
            "marginTop": 21.97265625,
          },
          undefined,
        ],
      ]
    }
  >
    Email
  </Text>
  <TextInput
    accessibilityLabel="input-email"
    autoCapitalize="none"
    correctText="email@domain.com"
    incorrectText="email@domain@domain.com"
    keyboardType="email-address"
    maxLength={100}
    placeholder="Email"
    placeholderTextColor="#A9A8A3"
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 32.8125,
          "height": "100%",
          "paddingHorizontal": 43.9453125,
          "paddingTop": 43.9453125,
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ]
    }
    testID="input-email"
    value="email@domain.com"
  />
</View>
`;

exports[`CustomInput Email incorrect should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#D5D5D533",
        "borderBottomWidth": 2.34375,
        "borderColor": "#D5D5D5",
        "borderLeftWidth": 7.03125,
        "borderRadius": 75,
        "borderRightWidth": 2.34375,
        "borderTopWidth": 7.03125,
        "height": 117.1875,
        "paddingHorizontal": 18.75,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 28.125,
          "fontWeight": "400",
        },
        Array [
          Object {
            "color": "#A9A8A3",
            "marginLeft": 10.986328125,
            "marginTop": 21.97265625,
          },
          undefined,
        ],
      ]
    }
  >
    Email
  </Text>
  <TextInput
    accessibilityLabel="input-email"
    autoCapitalize="none"
    correctText="email@domain.com"
    incorrectText="email@domain@domain.com"
    keyboardType="email-address"
    maxLength={100}
    placeholder="Email"
    placeholderTextColor="#A9A8A3"
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 32.8125,
          "height": "100%",
          "paddingHorizontal": 43.9453125,
          "paddingTop": 43.9453125,
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ]
    }
    testID="input-email"
    value="email@domain@domain.com"
  />
</View>
`;

exports[`CustomInput Name correct should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#D5D5D533",
        "borderBottomWidth": 2.34375,
        "borderColor": "#D5D5D5",
        "borderLeftWidth": 7.03125,
        "borderRadius": 75,
        "borderRightWidth": 2.34375,
        "borderTopWidth": 7.03125,
        "height": 117.1875,
        "paddingHorizontal": 18.75,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 28.125,
          "fontWeight": "400",
        },
        Array [
          Object {
            "color": "#A9A8A3",
            "marginLeft": 10.986328125,
            "marginTop": 21.97265625,
          },
          undefined,
        ],
      ]
    }
  >
    Name
  </Text>
  <TextInput
    correctText="Test"
    incorrectText="Test1234"
    maxLength={50}
    placeholder="Name"
    placeholderTextColor="#A9A8A3"
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 32.8125,
          "height": "100%",
          "paddingHorizontal": 43.9453125,
          "paddingTop": 43.9453125,
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ]
    }
    value="Test"
  />
</View>
`;

exports[`CustomInput Name incorrect should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#D5D5D533",
        "borderBottomWidth": 2.34375,
        "borderColor": "#D5D5D5",
        "borderLeftWidth": 7.03125,
        "borderRadius": 75,
        "borderRightWidth": 2.34375,
        "borderTopWidth": 7.03125,
        "height": 117.1875,
        "paddingHorizontal": 18.75,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 28.125,
          "fontWeight": "400",
        },
        Array [
          Object {
            "color": "#A9A8A3",
            "marginLeft": 10.986328125,
            "marginTop": 21.97265625,
          },
          undefined,
        ],
      ]
    }
  >
    Name
  </Text>
  <TextInput
    correctText="Test"
    incorrectText="Test1234"
    maxLength={50}
    placeholder="Name"
    placeholderTextColor="#A9A8A3"
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 32.8125,
          "height": "100%",
          "paddingHorizontal": 43.9453125,
          "paddingTop": 43.9453125,
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ]
    }
    value="Test1234"
  />
</View>
`;

exports[`CustomInput New password correct should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#D5D5D533",
        "borderBottomWidth": 2.34375,
        "borderColor": "#D5D5D5",
        "borderLeftWidth": 7.03125,
        "borderRadius": 75,
        "borderRightWidth": 2.34375,
        "borderTopWidth": 7.03125,
        "height": 117.1875,
        "paddingHorizontal": 18.75,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 28.125,
          "fontWeight": "400",
        },
        Array [
          Object {
            "color": "#A9A8A3",
            "marginLeft": 10.986328125,
            "marginTop": 21.97265625,
          },
          undefined,
        ],
      ]
    }
  >
    New password
  </Text>
  <TextInput
    autoCapitalize="none"
    correctText="Test12345!"
    incorrectText="Test234"
    keyboardType="default"
    maxLength={100}
    multiline={false}
    placeholder="New password"
    placeholderTextColor="#A9A8A3"
    secureTextEntry={true}
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 32.8125,
          "height": "100%",
          "paddingHorizontal": 43.9453125,
          "paddingTop": 43.9453125,
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ]
    }
    textContentType="newPassword"
    value="Test12345!"
  />
</View>
`;

exports[`CustomInput New password incorrect should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#D5D5D533",
        "borderBottomWidth": 2.34375,
        "borderColor": "#D5D5D5",
        "borderLeftWidth": 7.03125,
        "borderRadius": 75,
        "borderRightWidth": 2.34375,
        "borderTopWidth": 7.03125,
        "height": 117.1875,
        "paddingHorizontal": 18.75,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 28.125,
          "fontWeight": "400",
        },
        Array [
          Object {
            "color": "#A9A8A3",
            "marginLeft": 10.986328125,
            "marginTop": 21.97265625,
          },
          undefined,
        ],
      ]
    }
  >
    New password
  </Text>
  <TextInput
    autoCapitalize="none"
    correctText="Test12345!"
    incorrectText="Test234"
    keyboardType="default"
    maxLength={100}
    multiline={false}
    placeholder="New password"
    placeholderTextColor="#A9A8A3"
    secureTextEntry={true}
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 32.8125,
          "height": "100%",
          "paddingHorizontal": 43.9453125,
          "paddingTop": 43.9453125,
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ]
    }
    textContentType="newPassword"
    value="Test234"
  />
</View>
`;
